### PR TITLE
Add cache_dir argument for get_calibration_dataset

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
Add `cache_dir` argument for `get_calibration_dataset` to forward to `load_dataset`.

### CODE REVIEW

File: optimum/intel/neural_compressor/quantization.py [hunk 1]
```diff
@@ -285,6 +285,7 @@ def get_calibration_dataset(
         preprocess_function: Optional[Callable] = None,
         preprocess_batch: bool = True,
         use_auth_token: bool = False,
+        cache_dir: Optional[str] = None,
```
Comment (AlexKoff88) [inline optimum/intel/neural_compressor/quantization.py:288]: Please remove it from INC part and keep it only for OpenVINO.

File: optimum/intel/openvino/quantization.py [hunk 2]
```diff
@@ -378,11 +379,17 @@ def get_calibration_dataset(
                 Whether the `preprocess_function` should be batched.
             use_auth_token (`bool`, defaults to `False`):
                 Whether to use the token generated when running `transformers-cli login`.
+            cache_dir (`str`, *optional*):
+                Directory to read/write data. Defaults to `"~/.cache/huggingface/datasets"`.
```
Comment (AlexKoff88) [inline optimum/intel/openvino/quantization.py:383]:
```suggestion
                Caching directory for a calibration dataset.
```

File: optimum/intel/neural_compressor/quantization.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply optimum/intel/neural_compressor/quantization.py]: done

File: optimum/intel/openvino/quantization.py (hunk 2 above)
Comment (AlexanderDokuchaev) [reply optimum/intel/openvino/quantization.py]: Changed

**Review by echarlaix — APPROVED** [review]
LGTM, thanks for your contribution @AlexanderDokuchaev !

**Pull request merged**